# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/printing/nsPrintJob.cpp
# Commit: 5cd20e839a73
# Full Hash: 5cd20e839a7322ad075dcc6d88585a28d36db87d
# Author: Jonathan Watt <jwatt@jwatt.org>
# Date: 2022-05-23 09:52:44
# Regressor Bug: 1770539
# File Overlap Count: 1
# Description:
#   Bug 1770539 p2 - Move nsPrintData::mShrinkToFit to nsPrintJob. r=emilio
#   
#   Depends on D146973
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D146974
# ==============================================================================

diff -r f6e9ace877bd -r 5cd20e839a73 layout/printing/nsPrintJob.cpp
--- a/layout/printing/nsPrintJob.cpp	Sun May 22 09:37:55 2022 +0000
+++ b/layout/printing/nsPrintJob.cpp	Sun May 22 09:37:56 2022 +0000
@@ -434,7 +434,7 @@
       !printData->mPrintObject->mDocument->GetRootElement())
     return NS_ERROR_GFX_PRINTER_STARTDOC;
 
-  mPrintSettings->GetShrinkToFit(&printData->mShrinkToFit);
+  mPrintSettings->GetShrinkToFit(&mShrinkToFit);
 
   bool printingViaParent =
       XRE_IsContentProcess() && StaticPrefs::print_print_via_parent();
@@ -831,7 +831,7 @@
 
   // Here is where we figure out if extra reflow for shrinking the content
   // is required.
-  if (printData->mShrinkToFit) {
+  if (mShrinkToFit) {
     printData->mShrinkRatio = printData->mPrintObject->mShrinkRatio;
 
     if (printData->mShrinkRatio < 0.998f) {
@@ -1136,7 +1136,7 @@
 
 void nsPrintJob::UpdateZoomRatio(nsPrintObject* aPO) {
   if (!aPO->mParent) {
-    if (mPrt->mShrinkToFit) {
+    if (mShrinkToFit) {
       aPO->mZoomRatio = mPrt->mShrinkRatio;
       // If we're actually going to scale (the factor is less than 1), we
       // reduce the scale factor slightly to avoid the possibility of floating
@@ -1183,7 +1183,7 @@
   // Then we walk the frame tree and look for the "xmost" frame
   // this is the frame where the right-hand side of the frame extends
   // the furthest
-  if (mPrt->mShrinkToFit && aDocumentIsTopLevel) {
+  if (mShrinkToFit && aDocumentIsTopLevel) {
     nsPageSequenceFrame* pageSeqFrame = aPO->mPresShell->GetPageSequenceFrame();
     NS_ENSURE_STATE(pageSeqFrame);
     aPO->mShrinkRatio = pageSeqFrame->GetSTFPercent();